sqlmesh/core/engine_adapter/logical_merge.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+import typing as t
+
+from sqlglot import exp, parse_one
+
+from sqlmesh.core.engine_adapter.base import EngineAdapter, TransactionType
+
+if t.TYPE_CHECKING:
+    from sqlmesh.core._typing import TableName
+    from sqlmesh.core.engine_adapter._typing import QueryOrDF
+
+    class Base(EngineAdapter):
+        ...
+
+else:
+
+    class Base:
+        ...
+
+
+class LogicalMergeMixin(Base):
+    def merge(
+        self,
+        target_table: TableName,
+        source_table: QueryOrDF,
+        columns_to_types: t.Dict[str, exp.DataType],
+        unique_key: t.Sequence[str],
+    ) -> None:
```

Comment:
Can you add a comment describing this flow in english? We create a temp table, do a delete where blah, ect.